### PR TITLE
Update RandLANet

### PR DESCRIPTION
**Pull request by sanskar107**
Implement RandLANet in torch and tf.
image

This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/454)

### CODE REVIEW

File: ml3d/torch/models/randlanet.py [hunk 1]
```diff
@@ -41,136 +19,137 @@ class RandLANet(BaseModel):
...
-            num_classes=19,  # Number of valid classes
+            num_neighbors=16,
+            num_layers=4,
+            num_points=4096 * 11,
+            num_classes=19,
             ignored_label_inds=[0],
             sub_sampling_ratio=[4, 4, 4, 4],
-            dim_input=3,  # 3 + feature_dimension.
-            dim_feature=8,
+            in_channels=3,  # 3 + feature_dimension.
```
Comment (cosama) [inline ml3d/torch/models/randlanet.py:33]: If I use `ml3d/configs/randlanet_s3dis.yml` and the checkpoint https://storage.googleapis.com/open3d-releases/model-zoo/randlanet_s3dis_202201071330utc.pth for semantic segmentation I get:

```
RuntimeError: Error(s) in loading state_dict for RandLANet:
	size mismatch for fc0.weight: copying a param with shape torch.Size([8, 6]) from checkpoint, the shape in current model is torch.Size([8, 3]).
```

Looks like there is a discrepancy between the checkpoint file and the model.

```suggestion
            in_channels=6,  # 6 + feature_dimension.
```

Comment (sanskar107) [reply ml3d/torch/models/randlanet.py:33]: `in_channels` is already `6` in `ml3d/configs/randlanet_s3dis.yml`.
Refer https://github.com/isl-org/Open3D-ML/blob/sanskar/randlanet/ml3d/configs/randlanet_s3dis.yml#L22
I guess you are not loading the config file properly.

Comment (cosama) [reply ml3d/torch/models/randlanet.py:33]: You are right somehow my config was wrong Not sure how that happened. Looks all good. Sorry for that.

**Review by ssheorey — COMMENTED** [review]
Future PR: Add license header to all files, and add check for license header to CI. See main repo.

> Reviewed 8 of 13 files at r1, 3 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/454#-:-MtAP9Ru2rEjvlQWoXZ6:bc5bxcu)* status: 11 of 15 files reviewed, 9 unresolved discussions (waiting on @cosama, @sanskar107, and @ssheorey)

___
*[ml3d/configs/randlanet_parislille3d.yml, line 40 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAKd_W1pesQ5Nc44Fe:-MtAKd_W1pesQ5Nc44Ff:bnbg0mu) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_parislille3d.yml#L40)):*
> ```YAML
>   adam_lr: 0.01
>   batch_size: 2
>   learning_rate: 0.01
> ```

Use new hierarchical config format (`optimizer` -> `lr`) like in SparseConvUNet
Are `adam_lr` and `learning_rate` duplicates?
___
*[ml3d/configs/randlanet_parislille3d.yml, line 42 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAKRIC4Xj1hDsn5Xwt:-MtAKRIC4Xj1hDsn5Xwu:bediksu) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_parislille3d.yml#L42)):*
> ```YAML
>   learning_rate: 0.01
>   main_log_dir: ./logs
>   max_epoch: 100
> ```

Can you confirm that 100 epochs are enough now instead of 200?
___
*[ml3d/configs/randlanet_parislille3d.yml, line 54 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAM3NEDEKBJVfxKseZ:-MtAM3NEDEKBJVfxKse_:bha7av3) ([raw file](https://github.com/isl-org/open3d-ml/blob/6f12efc12fb172cb5878ea91f9c7606452bdbc06/ml3d/configs/randlanet_parislille3d.yml#L54)):*
> ```YAML
>     noise_level: 0.01
>     min_s: 0.9
>     max_s: 1.1
> ```

Are we reducing augmentation?
___
*[ml3d/configs/randlanet_s3dis.yml, line 38 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAMK2q8Mw9aAfmD6UC:-MtAMK2q8Mw9aAfmD6UD:bs22o05) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_s3dis.yml#L38)):*
> ```YAML
> pipeline:
>   name: SemanticSegmentation
>   adam_lr: 0.01
> ```

optimizer -> lr
___
*[ml3d/configs/randlanet_semantic3d.yml, line 52 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAMeT4Bd1l1nHP2gtI:-MtAMeT4Bd1l1nHP2gtJ:b-vjkovm) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_semantic3d.yml#L52)):*
> ```YAML
>   max_epoch: 100
>   save_ckpt_freq: 5
>   scheduler_gamma: 0.9886
> ```

nit: Is this hand-tuned? Would `0.99` work just as well (it's simpler)?
___
*[ml3d/datasets/augment/augmentation.py, line 23 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtANlJr7AAw7eg0WcYh:-MtANlJr7AAw7eg0WcYi:b27y58) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/datasets/augment/augmentation.py#L23)):*
> ```Python
>         Args:
>             data: Pointcloud or features.
>             cfg: configuration dictionary.
> ```

Document config option used. e.g.: Key 'dim' specifies...
___
*[ml3d/datasets/augment/augmentation.py, line 75 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAObTmDX-6LVuVK2SK:-MtAObTmDX-6LVuVK2SL:b-ylmwbp) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/datasets/augment/augmentation.py#L75)):*
> ```Python
> 
>         """
>         if np.abs(pc[:, :2].mean()) > 1e-2:
> ```

Why should we skip the Z dimension check? 
___
*[tests/test_models.py, line 60 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAPoSi1x3llVwOwcBp:-MtAPoSi1x3llVwOwcBq:b-5kbbak) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/tests/test_models.py#L60)):*
> ```Python
>     out = net(inputs).detach().numpy()
> 
>     assert out.shape == (1, 5000, 10)
> ```

[Future PR OK] Can we verify the output values as well? This will guard against accidental changes to the code that actually affect the model. One option to do that is:
- Create new RNG with fixed seed and create inputs and weights from this RNG.
- Scale inference output to int32 range and quantize to int32.
- Calculate hash: https://stackoverflow.com/a/66376073 and compare with hash value stored in test.

**Review by cosama — APPROVED** [review]

**Review by ssheorey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/454)* status: 11 of 15 files reviewed, 31 unresolved discussions (waiting on @cosama, @sanskar107, and @ssheorey)

___
*[ml3d/tf/models/randlanet.py, line 18 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG1ozd-g6d3PExgzhs:-MtG1ozd-g6d3PExgzht:b-ou3nqh) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L18)):*
> ```Python
>     Based on the architecture
>     https://arxiv.org/abs/1911.11236#
>     """
> ```

Keep link to reference implementation.
Also, more details on this model would be useful here. e.g.: image of the architecture and a brief description. Perhaps copying the paper abstract can be a good overview.

To add an image, drag and drop the image to the PR description box and add the image URL with rst image directive. The image will show up in sphinx.
___
*[ml3d/tf/models/randlanet.py, line 61 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG4x3r9-r-dFfjjx7H:-MtG4x3s1QLnsfFJGNRR:b-v9baid) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L61)):*
> ```Python
>                                                       momentum=0.99,
>                                                       epsilon=1e-6)
>         self.lr0 = tf.keras.layers.LeakyReLU(0.2)
> ```

Can we group these 3 together? Perhaps add to encoder, or as separate input layer.



_Code quote:_
```Python
        self.fc0 = tf.keras.layers.Dense(cfg.dim_features)
        self.bn0 = tf.keras.layers.BatchNormalization(-1,
                                                      momentum=0.99,
                                                      epsilon=1e-6)
        self.lr0 = tf.keras.layers.LeakyReLU(0.2)
```
___
*[ml3d/tf/models/randlanet.py, line 76 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG486q4-5KRe-rFWp3:-MtG486q4-5KRe-rFWp4:bpvm6cm) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L76)):*
> ```Python
>             encoder_dim_list.append(dim_feature)
> 
>         # self.encoder = tf.keras.models.Sequential(self.encoder)
> ```

Remove commented code
___
*[ml3d/tf/models/randlanet.py, line 92 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG4Ip55STCOQ7DhBmT:-MtG4Ip55STCOQ7DhBmU:b-4bygje) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L92)):*
> ```Python
>             dim_feature = encoder_dim_list[-i - 2]
> 
>         # self.decoder = tf.keras.models.Sequential(self.decoder)
> ```

remove commented code
___
*[ml3d/tf/models/randlanet.py, line 160 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG6yjH3pDYgK2IUFEq:-MtG6yjH3pDYgK2IUFEr:b-4bi6zl) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L160)):*
> ```Python
>         Args:
>             outputs: logits
>             labels: labels
> ```

Update docstring to match function arguments. Add expected shape.
___
*[ml3d/tf/models/randlanet.py, line 285 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG7UDa4ly3vKIXPIGC:-MtG7UDa4ly3vKIXPIGD:b-o500s6) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L285)):*
> ```Python
>                     feat = np.concatenate([pc, feat], axis=1)
> 
>                 assert cfg.in_channels == feat.shape[
> ```

`assert` -> `raise RuntimeError`
___
*[ml3d/tf/models/randlanet.py, line 326 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG7fIIEVsGXzpHpV1z:-MtG7fIIEVsGXzpHpV2-:b-o500s6) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L326)):*
> ```Python
>             feat = np.concatenate([pc, feat], axis=1)
> 
>         assert self.cfg.in_channels == feat.shape[
> ```

`assert` -> `raise RuntimeError`
___
*[ml3d/tf/models/randlanet.py, line 392 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG7rL01cIu47QAuQYE:-MtG7rL01cIu47QAuQYF:bpvm6cm) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L392)):*
> ```Python
>         # input_list += [feat, label]
> 
>         # return input_list
> ```

Remove commented code



_Code quote:_
```Python
        # input_list = input_points + input_neighbors + input_pools + input_up_samples
        # input_list += [feat, label]

        # return input_list
```
___
*[ml3d/tf/models/randlanet.py, line 551 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG87e9DEYNAFpYjti_:-MtG87e9DEYNAFpYjtia:b-4bygje) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L551)):*
> ```Python
>         self.encode_pos = encode_pos
> 
>         # self.device = device
> ```

remove commented code
___
*[ml3d/tf/models/randlanet.py, line 589 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLDkK85x_-mq8rmUwJ:-MtLDkK85x_-mq8rmUwK:bqq72hx) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L589)):*
> ```Python
>                 tf.Tensor of shape (B, d, N, 1)
>             neighbor_indices: indices of k neighbours.
>                 tf.Tensor of shape (B, N, K)
> ```

Add docs for `training` and `relative_features`
___
*[ml3d/torch/models/randlanet.py, line 22 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLFDItDh-hl6XdUhl_:-MtLFDIu1NJmBoXU4wze:b-ek91ru) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L22)):*
> ```Python
>     Based on the architecture
>     https://arxiv.org/abs/1911.11236#
>     """
> ```

[Same as for the TF version] keep ref. implementation link and add description + architecture image from paper.
___
*[ml3d/torch/models/randlanet.py, line 63 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLFTEbD5TBki3DTPx3:-MtLFTEbD5TBki3DTPx4:b-1yzos) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L63)):*
> ```Python
> 
>         self.fc0 = nn.Linear(cfg.in_channels, cfg.dim_features)
>         self.bn0 = nn.BatchNorm2d(cfg.dim_features, eps=1e-6, momentum=0.01)
> ```

[save as TF version] group into one layer if possible.
___
*[ml3d/torch/models/randlanet.py, line 177 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLIOiYCn_vQ7cBDg3I:-MtLIOiYCn_vQ7cBDg3J:bi25nyy) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L177)):*
> ```Python
>         if 'recenter' in augment_cfg:
>             val_augment_cfg['recenter'] = augment_cfg['recenter']
>             augment_cfg.pop('recenter')
> ```

`val_augment_cfg['recenter'] = augment_cfg.pop('recenter')`



_Code quote:_
```Python
            val_augment_cfg['recenter'] = augment_cfg['recenter']
            augment_cfg.pop('recenter')
```
___
*[ml3d/torch/models/randlanet.py, line 180 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLI_4LECS-FfxpTxvB:-MtLI_4LECS-FfxpTxvC:b-4noy7h) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L180)):*
> ```Python
>         if 'normalize' in augment_cfg:
>             val_augment_cfg['normalize'] = augment_cfg['normalize']
>             augment_cfg.pop('normalize')
> ```

same^^^
___
*[ml3d/torch/models/randlanet.py, line 196 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLIl8w6yqrh9FKiimd:-MtLIl8w6yqrh9FKiime:b24slk6) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L196)):*
> ```Python
>             feat = np.concatenate([pc, feat], axis=1)
> 
>         assert cfg.in_channels == feat.shape[
> ```

`assert` -> `raise`
___
*[ml3d/torch/models/randlanet.py, line 351 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLJMCB4HkuQqCCGSsQ:-MtLJMCB4HkuQqCCGSsR:b-uxfsj7) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L351)):*
> ```Python
>         Args:
>             outputs: logits
>             labels: labels
> ```

update docstring with shape and other args (+description as appropriate)
___
*[ml3d/torch/models/randlanet.py, line 426 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLJWugEDb269IIvBJO:-MtLJWugEDb269IIvBJP:bnsl4qx) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L426)):*
> ```Python
>             return False
> 
>     def update_probs(self, inputs, results, test_probs, test_labels):
> ```

Add docstring.
___
*[ml3d/torch/models/randlanet.py, line 447 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLJaKu7q0H37YWRoiY:-MtLJaKu7q0H37YWRoiZ:bn91ff6) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L447)):*
> ```Python
> 
> 
> class SharedMLP(nn.Module):
> ```

Add docstring: purpose of this class
___
*[ml3d/torch/models/randlanet.py, line 494 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLK4aHEaLeGshIQgTe:-MtLK4aI02kR5L3FCc3f:b8xm94r) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L494)):*
> ```Python
> 
> 
> class LocalSpatialEncoding(nn.Module):
> ```

docstring: purpose of class
___
*[ml3d/torch/models/randlanet.py, line 503 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLJvtIAwVz6hU3USdL:-MtLJvtJ1Ix82iblFee_:b-4bygje) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L503)):*
> ```Python
>         self.encode_pos = encode_pos
> 
>         # self.device = device
> ```

remove commented code
___
*[ml3d/torch/models/randlanet.py, line 578 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLK1VX9ZVuTV0dQ4S0:-MtLK1VX9ZVuTV0dQ4S1:b-76acrx) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L578)):*
> ```Python
> 
> 
> class AttentivePooling(nn.Module):
> ```

docstring: purpose of class.
___
*[ml3d/torch/models/randlanet.py, line 609 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLK82V-LSOD0zV7-NX:-MtLK82V-LSOD0zV7-NY:b8xm94r) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L609)):*
> ```Python
> 
> 
> class LocalFeatureAggregation(nn.Module):
> ```

docstring: purpose of class
___
*[ml3d/torch/models/randlanet.py, line 644 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtLKuooDCFrSqzfDL1_:-MtLKuooDCFrSqzfDL1a:b7rjvag) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/torch/models/randlanet.py#L644)):*
> ```Python
> 
>         """
>         # knn_output = knn(coords.cpu().contiguous(), coords.cpu().contiguous(), self.num_neighbors)
> ```

remove commented code.

**Comment by ssheorey** [comment]
Thank @sanskar107 for the updates. Could you address the lgtm alerts as well?

**Review by ssheorey — APPROVED** [review]
> Reviewed 2 of 13 files at r1, 2 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/454)* status: 7 of 15 files reviewed, 1 unresolved discussion (waiting on @sanskar107 and @ssheorey)

___
*[ml3d/configs/randlanet_parislille3d.yml, line 42 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAKRIC4Xj1hDsn5Xwt:-MtnNptUDTAflzwJlG_a:bbr0hd3) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_parislille3d.yml#L42)):*
<details><summary><i>Previously, sanskar107 (Sanskar Agrawal) wrote…</i></summary><blockquote>

Actually the training never converged on ParisLille3d. Not now, and not the last time. We converted the author's weights. But usually 100 epochs should suffice.
</blockquote></details>

OK. We should mention this in the main readme table (with a `(*) Using weights from original author`, for example). Users may likely run into this reproducibility issue as well.
___
*[ml3d/configs/randlanet_parislille3d.yml, line 54 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAM3NEDEKBJVfxKseZ:-MtnOm9c6_BwgQdniNfw:bnk8mf7) ([raw file](https://github.com/isl-org/open3d-ml/blob/6f12efc12fb172cb5878ea91f9c7606452bdbc06/ml3d/configs/randlanet_parislille3d.yml#L54)):*
<details><summary><i>Previously, sanskar107 (Sanskar Agrawal) wrote…</i></summary><blockquote>

Here `turn_on` is false, so none of the augmentations were being used. I tried our some from our new augmentation class, but that didn't help much. They tend to marginally lower the performance.
</blockquote></details>

OK. Perhaps then change to `dim: []` Is `turn_on` a default key? I don't see it in this file.
___
*[ml3d/configs/randlanet_semantic3d.yml, line 52 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAMeT4Bd1l1nHP2gtI:-MtnPd8L3IE1-_tp9U2T:bfijewv) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_semantic3d.yml#L52)):*
<details><summary><i>Previously, sanskar107 (Sanskar Agrawal) wrote…</i></summary><blockquote>

We picked this value from author's code, and I believe this might be hand tuned.
After 100 epochs, 0.99 will make learning rate 36% of initial value, while 0.9886 will make 30%. It shouldn't make any difference. Should I change it to 0.99 ?
</blockquote></details>

Don't worry about it - it will affect the weights. Going with the author's settings sounds good.
___
*[ml3d/datasets/augment/augmentation.py, line 75 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAObTmDX-6LVuVK2SK:-MtnQGUhALarWI1KRi-b:b9cprz6) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/datasets/augment/augmentation.py#L75)):*
<details><summary><i>Previously, sanskar107 (Sanskar Agrawal) wrote…</i></summary><blockquote>

It sometime helps to preserve the absolute height dimension. Usually height value is not very high and doesn't affect much if we rotate without recentering.
</blockquote></details>

Good idea. Can you add this as a comment?
___
*[ml3d/tf/models/randlanet.py, line 61 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtG4x3r9-r-dFfjjx7H:-MtnQvJ9CbUxSKI12VGF:bqowqlp) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/tf/models/randlanet.py#L61)):*
<details><summary><i>Previously, sanskar107 (Sanskar Agrawal) wrote…</i></summary><blockquote>

We can group them, I kept it similar to author's code. But is it critical ? The problem is that it will change name in the model state dict. Either we have to train on all datasets again, or manually strip down weights file and change the layer name.
</blockquote></details>

OK, not critical.

**Review by sanskar107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/454)* status: 6 of 15 files reviewed, 1 unresolved discussion (waiting on @ssheorey)

___
*[ml3d/configs/randlanet_parislille3d.yml, line 42 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAKRIC4Xj1hDsn5Xwt:-Mu0ztZXBZ3zWGDdH88G:bb6un1j) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/configs/randlanet_parislille3d.yml#L42)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

OK. We should mention this in the main readme table (with a `(*) Using weights from original author`, for example). Users may likely run into this reproducibility issue as well.
</blockquote></details>

Done
___
*[ml3d/configs/randlanet_parislille3d.yml, line 54 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAM3NEDEKBJVfxKseZ:-Mu0zetk2u4gmlhPWrzT:b-2vqc9l) ([raw file](https://github.com/isl-org/open3d-ml/blob/6f12efc12fb172cb5878ea91f9c7606452bdbc06/ml3d/configs/randlanet_parislille3d.yml#L54)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

OK. Perhaps then change to `dim: []` Is `turn_on` a default key? I don't see it in this file.
</blockquote></details>

`turn_on` was an ON/OFF check used in old augmentation methods. Not used anymore.
___
*[ml3d/datasets/augment/augmentation.py, line 75 at r2](https://reviewable.io/reviews/isl-org/open3d-ml/454#-MtAObTmDX-6LVuVK2SK:-Mu0zx-04vS-OXX5wT7A:b-896fix) ([raw file](https://github.com/isl-org/open3d-ml/blob/3bfa3bca288b275016aa9ba6f691ae87e4c08eb8/ml3d/datasets/augment/augmentation.py#L75)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

Good idea. Can you add this as a comment?
</blockquote></details>

Done.

**Pull request merged**